Library/Homebrew/dependency.rb

```diff
@@ -11,13 +11,20 @@ class Dependency
 
   DEFAULT_ENV_PROC = proc {}.freeze
 
-  def initialize(name, tags = [], env_proc = DEFAULT_ENV_PROC, option_names = [name])
+  def initialize(name, tags = [], env_proc = nil, option_names = [name])
     raise ArgumentError, "Dependency must have a name!" unless name
 
     @name = name
     @tags = tags
-    @env_proc = env_proc
     @option_names = option_names
+
+    if name.match?(/^openjdk(@|$)/)
+      env_proc ||= proc do
+        ENV["JAVA_HOME"] = to_formula.opt_libexec/"openjdk.jdk/Contents/Home"
```

Comment:
I don't think this path is valid on linux. There does not seem to be any `/home/linuxbrew/.linuxbrew/opt/openjdk/libexec` directory. Maybe this would need to be set to `to_formula.opt_prefix` on linux.

Recently bumped into an issue: https://github.com/Homebrew/linuxbrew-core/commit/e09613c1393abc7d07ed203c89d731500f5ad7aa

Reply:
I think `opt_prefix` actually also works on macOS.